addons/mrp/models/mrp_workorder.py

```diff
@@ -220,6 +221,14 @@ def _compute_duration(self):
             else:
                 order.duration_percent = 0
 
+    @api.depends('duration', 'duration_expected', 'state')
+    def _compute_progress(self):
+        for order in self:
+            if order.state == 'done':
+                order.progress = 100
+            elif order.duration_expected:
+                order.progress = order.duration * 100 / order.duration_expected
```

Comment:
rebase and add a default value for the progress (else: progress = 0)

Reply:
Okay.